tools/circt-lec/LogicExporter.cpp

```diff
@@ -0,0 +1,401 @@
+//===- LogicExporter.cpp - class to extrapolate CIRCT IR logic --*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// This file defines the logic-exporting class for the `circt-lec` tool.
+///
+//===----------------------------------------------------------------------===//
+
+#include "LogicExporter.h"
+#include "Circuit.h"
+#include "Solver.h"
+#include "Utility.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+#define DEBUG_TYPE "lec-exporter"
+
+namespace {
+/// Helper function to provide a common debug formatting for
+/// an operation's list of operands.
+template <class ConcreteOp>
+static void debugOperands(ConcreteOp op) {
+  for (const mlir::OpOperand &operand : op->getOpOperands()) {
+    mlir::Value value = operand.get();
+    lec::dbgs() << "Operand:\n";
+    lec::Scope indent;
+    lec::printValue(value);
+  }
+}
+
+/// Helper function to provide a common debug formatting for
+/// an operation's result.
+static void debugOpResult(const mlir::Value &result) {
```

Comment:
nit: const reference is not necessary for values/operations
```suggestion
static void debugOpResult(mlir::Value result) {
```